admin_manual/configuration_server/reverse_proxy_configuration.rst

```diff
@@ -50,6 +100,153 @@ the IP addresses of the proxy. This is useful when you use a reverse SSL proxy
+       rewrite ^ /index.php$uri;
+    }
+
+    location ~ ^/(?:build|tests|config|lib|3rdparty|templates|data)/ {
+       deny all;
+    }
+    location ~ ^/(?:\.|autotest|occ|issue|indie|db_|console) {
+       deny all;
+     }
+
+    location ~ ^/(?:index|remote|public|cron|core/ajax/update|status|ocs/v[12]|updater/.+|ocs-provider/.+|core/templates/40[34])\.php(?:$|/) {
+       include fastcgi_params;
+       fastcgi_split_path_info ^(.+\.php)(/.*)$;
+       fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
+       fastcgi_param PATH_INFO $fastcgi_path_info;
+       #Avoid sending the security headers twice
+       fastcgi_param modHeadersAvailable true;
+       fastcgi_param front_controller_active true;
+       fastcgi_pass unix:/run/php/php7.2-fpm.sock;
+       fastcgi_intercept_errors on;
+       fastcgi_request_buffering off;
+    }
+
+    location ~ ^/(?:updater|ocs-provider)(?:$|/) {
+       try_files $uri/ =404;
+       index index.php;
+    }
+
+    # Adding the cache control header for js and css files
+    # Make sure it is BELOW the PHP block
+    location ~* \.(?:css|js)$ {
+        try_files $uri /index.php$uri$is_args$args;
+        add_header Cache-Control "public, max-age=7200";
+        # Add headers to serve security related headers (It is intended to
+        # have those duplicated to the ones above)
+        add_header X-Content-Type-Options nosniff;
+        add_header X-Frame-Options "SAMEORIGIN";
+        add_header X-XSS-Protection "1; mode=block";
+        add_header X-Robots-Tag none;
+        add_header X-Download-Options noopen;
+        add_header X-Permitted-Cross-Domain-Policies none;
+        # Optional: Don't log access to assets
+        access_log off;
+   }
+
+   location ~* \.(?:svg|gif|png|html|ttf|woff|ico|jpg|jpeg)$ {
+        try_files $uri /index.php$uri$is_args$args;
+        # Optional: Don't log access to other assets
+        access_log off;
+   }
+ }
+
+For the config.php portion we have used the following configuration:
+
+::
+
+ <?php
+  $CONFIG = array (
+ ...
+  'trusted_domains' =>
+  array (
+    0 => 'nextcloud.domain.com',
+  ),
+  'trusted_proxies'   => ['IP_OF_REVERSE_PROXY'],
+  'overwrite.cli.url' => 'https://nextcloud.domain.com',
+  'overwritehost'     => 'nextcloud.domain',
+  'overwriteprotocol' => 'https',
+   ...
+    );
+
+Please take note that most details written in the config.php file will be done by Nextcloud itself but it is important to add:
+
+::
+
+  'trusted_proxies'   => ['IP_OF_REVERSE_PROXY'],
+  'overwrite.cli.url' => 'https://nextcloud.domain.com',
+  'overwritehost'     => 'nextcloud.domain',
+  'overwriteprotocol' => 'https',
+
+This will set for Nextcloud to trust the reverse proxy and properly function. Because we do not have additional websites or Nextcloud websites running on the same server we do not have to set the specific directory:
+
+::
+
+    'overwritewebroot'  => '/domain.tld/nextcloud',
+    'overwritecondaddr' => '^10\.0\.0\.1$',
```

Comment:
In general those two options are not needed for a reverse proxy setup where reverse proxy and proxied web server use different roots (like the reverse proxy use `/` and the actual web server serves from `/nextcloud`). Either highlight this in the text above or don't mention it.

Reply:
Ok I will make the requested changes.